VideoApp/VideoApp/Video/VideoAppCameraSource.swift

```diff
@@ -0,0 +1,168 @@
+    }
+    
+    deinit {
+        cameraSource?.stopCapture() // Prevent leaking the CameraSource when a Track/Source has been created
+    }
+
+    func destroyLocalVideoTrack() {
+        cameraSource?.stopCapture() { [weak self] _ in
+            self?.localVideoTrack = nil
+            self?.cameraSource = nil
+        }
+    }
+    
+    func flip() {
+        guard
+            let cameraSource = cameraSource,
+            let position = cameraSource.device?.position,
+            let captureDevice = CameraSource.captureDevice(position: position == .front ? .back : .front)
+            else {
+                return
+        }
+        
+        cameraSource.selectCaptureDevice(captureDevice) { [weak self] _, _, error in
+            guard error == nil else { return }
+            
+            self?.localMediaController.videoCaptureStarted()
+        }
+    }
+
+    func shouldMirrorLocalVideoView() -> Bool {
+        localVideoTrack?.shouldMirror ?? false
+    }
+
+    private func createCameraSource() {
+        let options = CameraSourceOptions() { builder in
+            if #available(iOS 13, *) {
+                builder.orientationTracker = UserInterfaceTracker(scene: UIApplication.shared.keyWindow!.windowScene!)
+            }
+            
+            switch self.appSettingsStore.topology {
+            case .group:
+                // Take a best guess and remove rotation tags using hardware acceleration
+                builder.rotationTags = .remove
+            case .peerToPeer:
+                break
+            }
+        }
+        
+        cameraSource = CameraSource(options: options, delegate: self)
+        
+        if let cameraSource = cameraSource {
+            localVideoTrack = LocalVideoTrack(source: cameraSource, enabled: true, name: "camera")
+            startCameraSource()
+        }
+    }
+    
+    private func startCameraSource() {
+        guard let cameraSource = cameraSource else { return }
+
+        let targetSize: CMVideoDimensions
+        let cropRatio: CGFloat
+        let frameRate: UInt
+        
+        switch appSettingsStore.videoCodec {
+        case .h264, .vp8:
+            // 640 x 480 squarish crop (1.13:1)
+            targetSize = CMVideoDimensions(width: 544, height: 480)
+            
+            cropRatio = CGFloat(targetSize.width) / CGFloat(targetSize.height)
+            frameRate = 20
+        case .vp8Simulcast:
+            // 1024 x 768 squarish crop (1.25:1) on most iPhones. 1280 x 720 squarish crop (1.25:1) on the iPhone X
+            // and models that don't have 1024 x 768.
+            targetSize = CMVideoDimensions(width: 900, height: 720)
+            
+            cropRatio = CGFloat(targetSize.width) / CGFloat(targetSize.height)
+            frameRate = 24 // With simulcast enabled there are 3 temporal layers, allowing a frame rate of f/4
+        }
+        
+        let captureDevice = CameraSource.captureDevice(position: .front)!
```

Comment:
It will crash and I was thinking that was ok for now given current architecture doesn't handle errors great and this should never be nil. But maybe it is possible? Perhaps the simulator? I'm pretty sure I tested on simulator.

I could do equivalent of objc and just put a guard at the top of the function that returns if nil.

Reply:
I moved this to a guard at the top of the function since there was no problem with doing so.